demos/device_shadow_for_aws/shadow_demo_main.c

```diff
@@ -581,191 +595,214 @@ int RunDeviceShadowDemo( bool awsIotMqttMode,
-                                                SHADOW_TOPIC_LENGTH_UPDATE_DELTA( THING_NAME_LENGTH ) );
+                xDemoStatus = UnsubscribeFromTopic( &xMqttContext,
+                                                    SHADOW_TOPIC_STRING_UPDATE_DELTA( THING_NAME ),
+                                                    SHADOW_TOPIC_LENGTH_UPDATE_DELTA( THING_NAME_LENGTH ) );
 
-            if( xDemoStatus != pdPASS )
-            {
-                LogError( ( "Failed to unsubscribe the topic %s",
-                            SHADOW_TOPIC_STRING_UPDATE_DELTA( THING_NAME ) ) );
+                if( xDemoStatus != pdPASS )
+                {
+                    LogError( ( "Failed to unsubscribe the topic %s",
+                                SHADOW_TOPIC_STRING_UPDATE_DELTA( THING_NAME ) ) );
+                }
             }
-        }
 
-        if( xDemoStatus == pdPASS )
-        {
-            xDemoStatus = UnsubscribeFromTopic( &xMqttContext,
-                                                SHADOW_TOPIC_STRING_UPDATE_ACCEPTED( THING_NAME ),
-                                                SHADOW_TOPIC_LENGTH_UPDATE_ACCEPTED( THING_NAME_LENGTH ) );
+            if( xDemoStatus == pdPASS )
+            {
+                xDemoStatus = UnsubscribeFromTopic( &xMqttContext,
+                                                    SHADOW_TOPIC_STRING_UPDATE_ACCEPTED( THING_NAME ),
+                                                    SHADOW_TOPIC_LENGTH_UPDATE_ACCEPTED( THING_NAME_LENGTH ) );
+
+                if( xDemoStatus != pdPASS )
+                {
+                    LogError( ( "Failed to unsubscribe the topic %s",
+                                SHADOW_TOPIC_STRING_UPDATE_ACCEPTED( THING_NAME ) ) );
+                }
+            }
 
-            if( xDemoStatus != pdPASS )
+            if( xDemoStatus == pdPASS )
             {
-                LogError( ( "Failed to unsubscribe the topic %s",
-                            SHADOW_TOPIC_STRING_UPDATE_ACCEPTED( THING_NAME ) ) );
+                xDemoStatus = UnsubscribeFromTopic( &xMqttContext,
+                                                    SHADOW_TOPIC_STRING_UPDATE_REJECTED( THING_NAME ),
+                                                    SHADOW_TOPIC_LENGTH_UPDATE_REJECTED( THING_NAME_LENGTH ) );
+
+                if( xDemoStatus != pdPASS )
+                {
+                    LogError( ( "Failed to unsubscribe the topic %s",
+                                SHADOW_TOPIC_STRING_UPDATE_REJECTED( THING_NAME ) ) );
+                }
             }
-        }
 
-        if( xDemoStatus == pdPASS )
-        {
-            xDemoStatus = UnsubscribeFromTopic( &xMqttContext,
-                                                SHADOW_TOPIC_STRING_UPDATE_REJECTED( THING_NAME ),
-                                                SHADOW_TOPIC_LENGTH_UPDATE_REJECTED( THING_NAME_LENGTH ) );
+            /* The MQTT session is always disconnected, even there were prior failures. */
```

Comment:
grammar: _"...even __if__ there were prior failures."_

Reply:
Thanks. Will update it.

Reply:
Fixed.